admin_ui/backend/api/config.py

```
@@ -3075,3 +3075,471 @@ async def delete_google_calendar_credentials(filename: str):
+        calendars_payload = _ms_graph_request_with_token(access_token, "GET", "/me/calendars")
+        calendars = calendars_payload.get("value") or []
+        username = (
+            (result.get("id_token_claims") or {}).get("preferred_username")
+            or me.get("userPrincipalName")
+            or me.get("mail")
+            or ""
+        )
+        with _ms_flow_lock():
+            _MS_DEVICE_FLOWS[flow_id]["status"] = "success"
+            _MS_DEVICE_FLOWS[flow_id]["result"] = {
+                "token_cache_path": token_cache_path,
```

Ruff (0.15.11)

[error] Undefined name `token_cache_path`

(F821)